common/include/pcl/PolygonMesh.h

```diff
@@ -23,6 +24,32 @@ namespace pcl
 
     std::vector< ::pcl::Vertices>  polygons;
 
+    /** \brief Add another polygon mesh to the current mesh.
+        * \param[in] rhs the mesh to add to the current mesh
+        * \return the new mesh as a concatenation of the current mesh and the new given mesh
+        */
+      inline PolygonMesh&
+      operator += (const PolygonMesh& rhs)
+      {
+        // Make the resultant polygon mesh take the newest stamp
+        header.stamp = std::max(header.stamp, rhs.header.stamp);
+
+        polygons.insert(polygons.end(), rhs.polygons.begin(), rhs.polygons.end());
```

Comment:
Don't we need to remap point indices in the RHS polygon?

Reply:
I don't use PolygonMesh but thought it was a simple enough addition to make. Could you please elaborate why we need to remap RHS polygon? It should be `const` since that's the assumption anyone would make. Did you mean the current mesh instead?

Reply:
Polygon mesh has two major components: points and faces. A face is defined by a set of points that belong to it. For efficiency reasons, instead of storing point coordinates, faces rather contain indices into the global points set. Now when you add two meshes, the indices of the second mesh need to be updated, otherwise they will point to the the points of the first mesh.

Reply:
I see (and I should have seen that while updating the `cloud`). In that case, RHS need not be updated, only the current mesh since it's the current cloud which gets updated. Will update the code for that. 👍

Reply:
Sorry, I should have used more precise language. RHS is `const`, so it's not updated of course. The polygons that are copied over from RHS need to be updated. I guess `std::transform` with `back_inserted` will be appropriate here.

Reply:
I modified the logic. Doesn't look really beautiful to me.